configure.py

```diff
@@ -760,8 +760,11 @@ def find_ninja():
 arg_parser.add_argument('--coverage', action = 'store_true', help = 'Compile scylla with coverage instrumentation')
 arg_parser.add_argument('--build-dir', action='store', default='build',
                         help='Build directory path')
-
+arg_parser.add_argument('-h', '--help', action='store_true', help='show this help message and exit')
 args = arg_parser.parse_args()
+if args.help:
+    arg_parser.print_help()
+    arg_parser.exit()
```

Comment:
yes, but the usage i want to address is like:
```console
$ ./configure.py --foobar --help
usage: Configure scylla [-h] [--out BUILDFILE] [--out-final-name BUILDFILE_FINAL_NAME] [--mode {debug,release,dev,sanitize,coverage}] [--with ARTIFACTS] [--with-seastar SEASTAR_PATH] [--enable-dist]
                        [--disable-dist] [--dist-only] [--cflags USER_CFLAGS] [--ldflags USER_LDFLAGS] [--target TARGET] [--compiler CXX] [--c-compiler CC] [--enable-dpdk] [--disable-dpdk]
                        [--dpdk-target DPDK_TARGET] [--debuginfo DEBUGINFO] [--optimization-level MODE=LEVEL] [--static-stdc++] [--static-boost] [--static-yaml-cpp] [--tests-debuginfo TESTS_DEBUGINFO]
                        [--perf-tests-debuginfo PERF_TESTS_DEBUGINFO] [--python PYTHON] [--split-dwarf] [--enable-alloc-failure-injector] [--enable-seastar-debug-allocations] [--with-antlr3 ANTLR3_EXEC]
                        [--with-ragel RAGEL_EXEC] [--enable-stack-guards] [--disable-stack-guards] [--verbose] [--test-repeat TEST_REPEAT] [--test-timeout TEST_TIMEOUT]
                        [--clang-inline-threshold CLANG_INLINE_THRESHOLD] [--list-artifacts] [--date-stamp DATE_STAMP] [--use-cmake] [--coverage] [--build-dir BUILD_DIR]
...
$ echo $?
0
```

`--help` is added so that the given call is always a noop.

`argparse` invokes the action registered by the added options, if the action calls `sys.exit()`. the code which checks for unknown args is not executed. briefly, it works like:
```py
    def parse_args(self, args=None, namespace=None):
        # set the value to the namespace to be returned, 
        # calls into the registered actions, and
        # consume the positional options, 
        args, argv = self.parse_known_args(args, namespace)
        if argv:
            msg = _('unrecognized arguments: %s') % ' '.join(argv)
            if self.exit_on_error:
                self.error(msg)
            else:
                raise ArgumentError(None, msg)
        return args
```